File: executor/src/main/java/com/codeloom/executor/engine/callbacks/SimpleLogCallback.java
package com.codeloom.executor.engine.callbacks;

import com.github.dockerjava.api.async.ResultCallbackTemplate;
import com.github.dockerjava.api.model.*;
import java.nio.charset.StandardCharsets;

public class SimpleLogCallback extends ResultCallbackTemplate<SimpleLogCallback, Frame> {
  private final StringBuilder out, err;

  public SimpleLogCallback(StringBuilder o, StringBuilder e) {
    out = o;
    err = e;
  }

  public void onNext(Frame f) {
    String m = new String(f.getPayload(), StandardCharsets.UTF_8);
    (f.getStreamType() == StreamType.STDERR ? err : out).append(m);
  }
}

File: executor/src/main/kotlin/com/codeloom/executor/service/SimpleLogCallback.kt
package com.codeloom.executor.service

import com.github.dockerjava.api.async.ResultCallbackTemplate
import com.github.dockerjava.api.model.Frame
import com.github.dockerjava.api.model.StreamType

class SimpleLogCallback(
    private val stdout: StringBuilder,
    private val stderr: StringBuilder,
) : ResultCallbackTemplate<SimpleLogCallback, Frame>() {
    override fun onNext(frame: Frame) {
        val message = String(frame.payload, Charsets.UTF_8)
        if (frame.streamType == StreamType.STDERR) {
            stderr.append(message)
        } else {
            stdout.append(message)
        }
    }
}